File: src/2529.maximum-count-of-positive-integer-and-negative-integer.2614/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumCount = function (nums) {
    let left = 0;
    let right = nums.length - 1;
    let firstPositive = nums.length;
    while (left <= right) {
        const mid = (left + right) >> 1;
        if (nums[mid] > 0) {
            firstPositive = Math.min(firstPositive, mid);
            right = mid - 1;
        } else {
            left = mid + 1;
        }
    }
    const positiveCount = nums.length - firstPositive;

    left = 0;
    right = firstPositive - 1;
    let lastNegative = -1;
    while (left <= right) {
        const mid = (left + right) >> 1;
        if (nums[mid] < 0) {
            lastNegative = Math.max(lastNegative, mid);
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    const negativeCount = lastNegative + 1;

    return Math.max(positiveCount, negativeCount);
};

File: src/2529.maximum-count-of-positive-integer-and-negative-integer.2614/solution.java
class Solution {
    public int maximumCount(int[] nums) {
        int firstPostiveIndex = nums.length;
        int lastNegativeIndex = -1;

        int l = 0;
        int h = nums.length - 1;
        while (l <= h) {
            int mid = (l + h) >> 1;
            if (nums[mid] > 0) {
                firstPostiveIndex = mid;
                h = mid - 1;
            } else {
                if (nums[mid] < 0) {
                    lastNegativeIndex = mid;
                }
                l = mid + 1;
            }
        }
        l = lastNegativeIndex + 1;
        while (l <= h) {
            int mid = (l + h) >> 1;
            if (nums[mid] < 0) {
                lastNegativeIndex = mid;
                l = mid + 1;
            } else {
                h = mid - 1;
            }
        }

        return Math.max(lastNegativeIndex + 1, nums.length - firstPostiveIndex);
    }
}